ivory-bsp-posix/src/Ivory/BSP/ARMv7A/RaspberryPi/WiringPi.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Ivory.BSP.ARMv7A.RaspberryPi.WiringPi where
+
+
+import Ivory.Language
+
+wiringPi_header :: String
+wiringPi_header = "wiringPi.h"
+
+softTone_header :: String
+softTone_header = "softTone.h"
+
+-------------
+-- TYPES FOR 32 BITS
+-------------
+
+type CSize_t = Uint32
+type CInt    = Sint32
+
+
+------------------
+--SETUP FUNCTIONS
+-- see http://wiringpi.com/reference/setup/
+-------------------
+
+-- Setup with the wiringPi numbering
```

Comment:
Could you make these sorts of comments into [Haddock](https://www.haskell.org/haddock/doc/html/ch03s08.html)?